src/app/WriteHandler.cpp

```diff
@@ -258,17 +258,19 @@ CHIP_ERROR WriteHandler::SendWriteResponse(System::PacketBufferTLVWriter && aMes
 
 void WriteHandler::DeliverListWriteBegin(const ConcreteAttributePath & aPath)
 {
-    if (auto * attrOverride = AttributeAccessInterfaceRegistry::Instance().Get(aPath.mEndpointId, aPath.mClusterId))
+    if (mDataModelProvider != nullptr)
     {
-        attrOverride->OnListWriteBegin(aPath);
+        mDataModelProvider->ListAttributeWriteNotification(aPath, DataModel::ListWriteOperation::kListWriteBegin);
     }
 }
 
 void WriteHandler::DeliverListWriteEnd(const ConcreteAttributePath & aPath, bool writeWasSuccessful)
 {
-    if (auto * attrOverride = AttributeAccessInterfaceRegistry::Instance().Get(aPath.mEndpointId, aPath.mClusterId))
+    if (mDataModelProvider != nullptr)
     {
-        attrOverride->OnListWriteEnd(aPath, writeWasSuccessful);
+        mDataModelProvider->ListAttributeWriteNotification(aPath,
+                                                           writeWasSuccessful ? DataModel::ListWriteOperation::kListWriteEndFinal
+                                                                              : DataModel::ListWriteOperation::kListWriteEnd);
```

Comment:
This does not accurately capture what happens now with the success/fail arg, right?

Reply:
Existing docs does not explain what `aWriteWasSuccessful` means ... does this mean that the `WriteAttribute` succeeded? Why is that important and why can't that be tracked by the AAI if it needs it?

Reply:
Digging a little in the past issues it seems that the intention of "aWriteWasSuccessful" is indeed to notify if the write went as intended (true) or if there were some issues with the operation (false) like timeouts or some other "abort" reasons.